rust/src/interpreter/utils.rs

```diff
@@ -0,0 +1,127 @@
+use std::cmp::min;
+
+use evmc_vm::{ExecutionMessage, MessageFlags, Revision, StatusCode};
+
+use crate::{interpreter::gas::consume_copy_cost, types::u256};
+
+#[inline(always)]
+pub fn get_mut_slice_within_bounds<T>(data: &[T], offset: u256, len: u64) -> &[T] {
+    if len == 0 {
+        return &[];
+    }
+    let (offset, offset_overflow) = offset.into_u64_with_overflow();
+    if offset_overflow {
+        return &[];
+    }
+    let offset = offset as usize;
+    let len = len as usize;
+    let (end, end_overflow) = offset.overflowing_add(len);
+    if end_overflow || offset >= data.len() {
+        &[]
+    } else {
+        &data[offset..min(end, data.len())]
+    }
+}
+
+#[inline(always)]
+pub fn zero_slice(data: &mut [u8]) {
+    for byte in data {
+        *byte = 0;
+    }
+}
+
+#[inline(always)]
+pub fn copy_slice_padded(
+    src: &[u8],
+    dest: &mut [u8],
+    gas_left: &mut u64,
+) -> Result<(), StatusCode> {
+    consume_copy_cost(gas_left, dest.len() as u64)?;
+    dest[..src.len()].copy_from_slice(src);
+    zero_slice(&mut dest[src.len()..]);
+    Ok(())
+}
+
+#[inline(always)]
+pub fn word_size(bytes: u64) -> Result<u64, StatusCode> {
```

Comment:
```suggestion
pub fn into_word_size(bytes: u64) -> Result<u64, StatusCode> {
```

Reply:
I think `byte_len` for the argument makes more sense. Will change that.